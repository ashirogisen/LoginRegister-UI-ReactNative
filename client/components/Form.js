import React, { useState, useEffect } from 'react'
import { StyleSheet } from 'react-native'
import { Item, Input, Icon } from 'native-base';

function Form(props) {
  const styles = StyleSheet.create({
    container: {
      width: '100%',
      margin: 10,
      backgroundColor: '#888888',
      color: 'white',
    },
    icon: {
      marginHorizontal: 18,
      color: props.iconColor,
      width: '10%',
      height: 20,
    },
  })

  return (
    <Item testID="test-formItem" rounded style={styles.container}>
      <Icon testID="test-formIcon" type="Ionicons" style={styles.icon} name={props.icon} size={23} active={false} />
      <Input 
        testID="test-formInput"
        onChangeText={v => props.setValue(v)} 
        placeholder={props.placeholder} 
        placeholderTextColor="#bdc3c7"
        autoCapitalize="none"
        keyboardType="default"
      />
    </Item>
  )
}

export default Form