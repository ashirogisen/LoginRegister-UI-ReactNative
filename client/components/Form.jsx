import React from 'react'
import { StyleSheet } from 'react-native'
import { Container, Header, Content, Item, Input } from 'native-base';

function Form() {
  const styles = StyleSheet.create({
    container: {
      width: '90%',
      height: '100%',
      flexDirection: 'row'
    }
  })

  return (
    <Container style={styles.container}>
      <Content>
        <Item rounded>
          <Input placeholder='Rounded Textbox'/>
        </Item>
      </Content>
    </Container>
  )
}

export default Form