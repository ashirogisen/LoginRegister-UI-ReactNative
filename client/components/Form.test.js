import React from 'react'
import Form from './Form'
import renderer from 'react-test-renderer';
import { render, fireEvent } from 'react-native-testing-library';

const onChangeTextMocked = jest.fn();
const CHANGE_TEXT = 'Test';

describe('Render Form', () => {
  it('Renders without crashing', () => {
    const tree = renderer.create(
      <Form />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('Should render a forms item/container', () => {
    const { debug, getByTestId } = render (
      <Form />
    )
    expect(getByTestId('test-formItem')).toBeTruthy()
  })

  it('Should render a forms icon', () => {
    const { getByTestId } = render (
      <Form />
    )
    expect(getByTestId('test-formIcon')).toBeTruthy()
  })

  it('Should render a forms input-box', () => {
    const { getByTestId } = render (
      <Form />
    )
    expect(getByTestId('test-formInput')).toBeTruthy()
  })

  it('setValue is valid', () => {
    const { getByTestId } = render (
      <Form setValue={onChangeTextMocked} />
    )
    fireEvent.changeText(getByTestId('test-formInput'), CHANGE_TEXT);
  })
})
