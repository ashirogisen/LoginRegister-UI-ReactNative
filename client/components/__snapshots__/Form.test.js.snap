// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form Renders without crashing 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#888888",
      "borderBottomWidth": 1,
      "borderColor": "#D9D5DC",
      "borderLeftWidth": 1,
      "borderRadius": 30,
      "borderRightWidth": 1,
      "borderTopWidth": 1,
      "borderWidth": 1,
      "color": "white",
      "flexDirection": "row",
      "margin": 10,
      "marginLeft": 2,
      "opacity": 1,
      "width": "100%",
    }
  }
  testID="test-formItem"
>
  <Text />
  <TextInput
    allowFontScaling={true}
    autoCapitalize="none"
    editable={true}
    keyboardType="default"
    onChangeText={[Function]}
    placeholderTextColor="#bdc3c7"
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#000",
        "flex": 1,
        "fontSize": 17,
        "height": 50,
        "paddingLeft": 8,
        "paddingRight": 5,
        "top": 1.5,
      }
    }
    testID="test-formInput"
    underlineColorAndroid="rgba(0,0,0,0)"
  />
</View>
`;
